web/packages/shared/hooks/useSlidingHover.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useRef } from 'react';
+
+// useSlidingHover provides a hook that can be used to create a background that slides between
+// elements on hover. It takes an argument for the delay in milliseconds for the transition.
+//
+// parentRef is for the parent element that contains children that will have their hover state
+// handled by this hook. The parent element should have position: relative.
+//
+// ref is for the sliding background element. It should have position: absolute.
+//
+// handleMouseLeave is for the onMouseLeave event handler for the parent element.
+// handleMouseEnter is for the onMouseEnter event handler for the children elements.
+//
+// Example usage:
+//
+// const {
+//   ref,
+//   parentRef,
+//   handleMouseLeave,
+//   handleMouseEnter,
+// } = useSlidingHover(200);
+//
+// return (
+//   <TabsContainer ref={parentRef} onMouseLeave={handleMouseLeave}>
+//     <Tab onMouseEnter={handleMouseEnter}>Item 1</div>
+//     <Tab onMouseEnter={handleMouseEnter}>Item 2</div>
+//     <Tab onMouseEnter={handleMouseEnter}>Item 3</div>
+//
+//     <TabBackground ref={ref} />
+//   </div>
```

Comment:
i tried the example but it didn't work for me 🤔 , should we create a working story to catalog it?

```suggestion
//     <Tab onMouseEnter={handleMouseEnter}>Item 1</Tab>
//     <Tab onMouseEnter={handleMouseEnter}>Item 2</Tab>
//     <Tab onMouseEnter={handleMouseEnter}>Item 3</Tab>
//
//     <TabBackground ref={ref} />
//   </TabsContainer>
```